### ISSUE TITLE
OOBE - Name your IoT device step failed with TPM error

### ISSUE THREAD

**Issue by Vivihung** [microsoft/Project-Santa-Cruz-Preview#115]
Following the [OOBE document](https://github.com/microsoft/Project-Santa-Cruz-Preview/blob/main/user-guides/getting_started/oobe.md#oobe-procedure), hit TPM error at step 13:

![image](https://user-images.githubusercontent.com/4814277/105561247-0fe95b00-5ccb-11eb-9404-1ff50d0c48a3.png)

If hit the "Next" button the 2nd time, the error changed to
![image](https://user-images.githubusercontent.com/4814277/105561280-25f71b80-5ccb-11eb-8c5a-0e28a71fd114.png)

Because of the error, the OOBE experience is blocked. Not able to finish it.

At this point, if go to the Azure IoT Edge portal, can see the device is created.
However, you might tell it seems not well configured due to the runtime is N/A and the module could is 0. (am expecting more than 1 modules installed)
![image](https://user-images.githubusercontent.com/4814277/105561448-930ab100-5ccb-11eb-8780-ef070961f26a.png)

What do I miss?

**Comment by Vivihung** [comment]
Note: 
Since the device is not configured correctly, I reconnect the device with Azure with the created device's connection string (got the connection string from Azure IoT extension of Azure CLI) via go over the OOBE again:
[The 2nd button]
![image](https://user-images.githubusercontent.com/4814277/105561787-acf8c380-5ccc-11eb-9c27-a3252b31f831.png)

The wizard told the configuration is completed
![image](https://user-images.githubusercontent.com/4814277/105561912-0e209700-5ccd-11eb-9891-13871190e1e1.png)

And I can see more modules in the Azure IoT Hub page
(Please ignore the device name is different than previous steps, I have tried it multiple times and delete/re-add devices with the same or different name. None of them succeed)
![image](https://user-images.githubusercontent.com/4814277/105561965-3ad4ae80-5ccd-11eb-8e3c-ecb6cb9868af.png)

I though the configuration is done correctly via the connection string setup.
However, while trying the nocode vision prototypes, during the step of take photo, the page throws error and complains about modules...
![image](https://user-images.githubusercontent.com/4814277/105562106-cf3f1100-5ccd-11eb-805c-18452fbc8210.png)

This potentially could be a different issue, but I would like a way to make sure the OOBE is completed.

**Comment by Vivihung** [comment]
I unblock myself by not checking the TPM enrollment checkbox. 
Finally see the setup complete UI!

This is what I see in the Azure IoT Hub portal
![image](https://user-images.githubusercontent.com/4814277/105562705-1e864100-5cd0-11eb-86dc-6410aa62de7e.png)

This brings up another question... why setup with existing device via connection string shows the setup complete but actually the  device does not looks in a healthy state?

Let me know if you need a separate issue to track it.

**Comment by WendyAnn0079** [comment]
Hello - thanks for reaching out - 
1. When you went the "Connect a New Device" route, did you select an IoT Hub that is linked to a DPS instance?  If you don't have an IoT Hub that is linked to a DPS instance, TPM enrollment will fail.  The TPM box should be greyed out so you don't have the option to select it.  I will investigate what is happening - thanks for bringing this to our attention. 
2. If you go the "Connect with an existing Connection String" route no modules are installed.  So everything is working as expected.

**Comment by Vivihung** [comment]
@WendyAnn0079 
1. No, the IoT Hub I selected didn't link to a DPS instance. It's a new IoT Hub that I just created for Santa Cruz. Since the onboarding document doesn't have any step for DPS, I believe my IoT Hub isn't linked to a DPS instance.
2. Do you mean this is expected? 
=>"Connect with an existing Connection String" will not install any module.

**Comment by WendyAnn0079** [comment]
@Vivihung 
1. Thank you, a bug has been filed and we will update the on-boarding documentation as well.  
2. Yes, going the Existing Connection String route will not install modules (as expected).  If you are going this route, it is expected that you have already installed the modules and you are just installing a new IoT device.

**Comment by v-prsasa** [comment]
Possibly due to an older build, internet connection or due to not selecting an IoT Hub with a DPS instance. Hence, closing this issue.
